test/Serializer.js

```diff
@@ -0,0 +1,251 @@
+		expect(results[0]).to.not.have.property("passwordHash");
+
+		expect(results[1]).to.have.property("id");
+		expect(results[1]).to.have.property("name");
+		expect(results[1]).to.have.property("email");
+		expect(results[1]).to.have.property("phone");
+		expect(results[1]).to.have.property("status");
+		expect(results[1]).to.not.have.property("passwordHash");
+
+		expect(noDocsResults).to.be.an("array").length(0);
+	});
+
+	it("Should accept an options object instead of a name", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const resultOne = docs[0].serialize(["phone", "status"]);
+		const resultTwo = docs[1].serialize({include: ["id", "name"]});
+		const resultThree = docs[1].serialize(7);
+		expect(resultOne).to.have.property("phone");
+		expect(resultOne).to.have.property("status");
+		expect(resultOne).to.not.have.property("id");
+		expect(resultOne).to.not.have.property("name");
+		expect(resultOne).to.not.have.property("email");
+		expect(resultOne).to.not.have.property("passwordHash");
+
+		expect(resultTwo).to.have.property("id");
+		expect(resultTwo).to.have.property("name");
+		expect(resultTwo).to.not.have.property("email");
+		expect(resultTwo).to.not.have.property("phone");
+		expect(resultTwo).to.not.have.property("passwordHash");
+		expect(resultTwo).to.not.have.property("status");
+
+		expect(Object.keys(resultThree)).to.be.an("array").length(0);
+	});
+
+	it("Should add all document fields to the output prior to running modify without include or exclude statements", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("isActive");
+		expect(result).to.have.property("id");
+		expect(result).to.have.property("name");
+		expect(result).to.have.property("email");
+		expect(result).to.have.property("phone");
+		expect(result).to.have.property("passwordHash");
+		expect(result).to.have.property("status");
+		expect(result).to.have.property("isActive");
+	});
+
+	it("Should add all document fields to the output prior to running both include and exclude statements", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("hideSecure");
+		expect(result).to.have.property("id");
+		expect(result).to.have.property("name");
+		expect(result).to.have.property("email");
+		expect(result).to.have.property("phone");
+		expect(result).to.have.property("status");
+		expect(result).to.not.have.property("passwordHash");
+	});
+
+	it("Should fail while serializing a document and return a blank object", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("nonExistingSerializer");
+		expect(Object.keys(result)).to.be.an("array").length(0);
+	});
+
+	it("Should throw errors on invalid usage", () => {
+		expect(() => {
+			User.serializer.add({invalidUsage: "Should fail"}, ["name", "email"]);
+		}).to.throw("Field name is required and should be of type string");
```

Comment:
How do we feel about making these one line?

```suggestion
		expect(() => User.serializer.add({invalidUsage: "Should fail"}, ["name", "email"])).to.throw("Field name is required and should be of type string");
```

Reply:
Fixed in 3ed4e6af6d0bf403bf997942918c7cf31140153f